plugins/c8y_configuration_plugin/src/download.rs

```diff
@@ -142,30 +140,45 @@ pub async fn handle_config_download_request_child_device(
         file_entry,
     };
 
-    let mut download_info = DownloadInfo::new(&smartrest_request.url);
-    // If the provided url is a c8y url, add auth
-    if http_client.url_is_in_my_tenant_domain(&smartrest_request.url) {
-        let token = http_client.get_jwt_token().await?;
-        download_info.auth = Some(Auth::new_bearer(&token.token()));
-    }
-
-    info!("Downloading config file update from Cumulocity url: {download_info:?}");
-    let config_file_name = config_type.as_str();
-
-    let path_to = PathBuf::from(format!(
-        "{FILE_TRANSFER_ROOT_PATH}/{}",
-        config_management.http_file_repository_relative_path()
-    ));
-    let temp_download_file_name = &format!(".tmp_{}", config_file_name);
-    let downloader = Downloader::new(temp_download_file_name, &None, FILE_TRANSFER_ROOT_PATH);
-    downloader.download(&download_info).await?;
-    downloader.rename(&path_to).await?;
+    if let Err(err) = download_config_file(
+        smartrest_request.url.as_str(),
+        config_management
+            .file_transfer_repository_full_path()
+            .into(),
+        tmp_dir,
+        PermissionEntry::new(None, None, None), //no need to change ownership of downloaded file
+        http_client,
+    )
+    .await
+    {
+        // Fail the operation in the cloud if the file download itself fails
+        // by sending EXECUTING and FAILED responses back to back
+
+        let c8y_child_topic =
+            Topic::new_unchecked(&C8yTopic::ChildSmartRestResponse(child_id).to_string());
+
+        let executing_msg = Message::new(
+            &c8y_child_topic,
+            DownloadConfigFileStatusMessage::status_executing()?,
+        );
+        mqtt_client.published.send(executing_msg).await?;
```

Comment:
This should be done before calling `download_config_file`. Isn't it?

And even before the line:

```
let file_entry = plugin_config.get_file_entry_from_type(&config_type)?;
```

Furthermore, if this line fails to find the requested file entry (say the device and the cloud are not in sync),
then a `failed` status must be sent to the cloud.

Reply:
> This should be done before calling download_config_file. Isn't it?

Implicitly sending an EXECUTING response, without expecting the child device to send it, is not possible with the current stateless plugin. C8y has this limitation that only a single EXECUTING response can be sent for an operation. If you send a second one, it would mark the next PENDING operation as EXECUTING instead of re-marking the original operation as EXECUTING (a side effect of the C8Y design that blindly associates operation responses to the **last** PENDING operation in the queue, without validating the config type in the payload, preventing idempotent response handling).

So, currently, we're marking an operation as EXECUTING in the cloud, only when a child device send the `"status": "executing"` response. To handle multiple EXECUTING responses, we'll have to make the plugin stateful which tracks if an operation was already marked EXECUTING or not.

Reply:
Okay, I understand. That might deserve a comment. Having an "executing" sent only on error is puzzling.